CMakeLists.txt

```diff
@@ -278,6 +278,12 @@ if (WIN32)
 	add_subdirectory(lib/luaproxy/)
 endif()
 
+if (CYGWIN)
+	add_definitions(-D_GNU_SOURCE)
+	set(M_PI 3.14159265358979323846264338327950288) # This should be moved elsewhere
```

Comment:
According to this, the proper way is to define `_USE_MATH_DEFINES`:
https://stackoverflow.com/questions/1727881/how-to-use-the-pi-constant-in-c#1727896